evalml/tests/component_tests/test_target_encoder.py

```diff
@@ -0,0 +1,147 @@
+        encoder.fit(X, y)
+
+
+def test_cols():
+    X = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                      'col_2': ['2', '1', '1', '1', '1'],
+                      'col_3': ["a", "a", "a", "a", "a"]})
+    y = pd.Series([0, 1, 1, 1, 0])
+    encoder = TargetEncoder(cols=[])
+    encoder.fit(X, y)
+    X_t = encoder.transform(X)
+    pd.testing.assert_frame_equal(X, X_t)
+
+    encoder = TargetEncoder(cols=['col_2'])
+    encoder.fit(X, y)
+    X_t = encoder.transform(X)
+    X_expected = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                               'col_2': [0.60000, 0.742886, 0.742886, 0.742886, 0.742886],
+                               'col_3': ["a", "a", "a", "a", "a"]})
+    pd.testing.assert_frame_equal(X_t, X_expected, check_less_precise=True)
+
+    encoder = TargetEncoder(cols=['col_2', 'col_3'])
+    encoder.fit(X, y)
+    X_t = encoder.transform(X)
+    encoder2 = TargetEncoder()
+    encoder2.fit(X, y)
+    X_t2 = encoder2.transform(X)
+    pd.testing.assert_frame_equal(X_t, X_t2)
+
+
+def test_transform():
+    X = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                      'col_2': ["r", "t", "s", "t", "t"],
+                      'col_3': ["a", "a", "a", "b", "a"]})
+    y = pd.Series([0, 1, 1, 1, 0])
+    encoder = TargetEncoder()
+    encoder.fit(X, y)
+    X_t = encoder.transform(X)
+    X_expected = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                               'col_2': [0.6, 0.65872, 0.6, 0.65872, 0.65872],
+                               'col_3': [0.504743, 0.504743, 0.504743, 0.6, 0.504743]})
+    pd.testing.assert_frame_equal(X_t, X_expected)
+
+
+def test_smoothing():
+    # larger smoothing values should bring the values closer to the global mean
+    X = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                      'col_2': [2, 1, 1, 1, 1],
+                      'col_3': ["a", "a", "a", "a", "b"]})
+    y = pd.Series([0, 1, 1, 1, 0])
+    encoder = TargetEncoder(smoothing=1)
+    encoder.fit(X, y)
+    X_t = encoder.transform(X)
+    X_expected = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                               'col_2': [2, 1, 1, 1, 1],
+                               'col_3': [0.742886, 0.742886, 0.742886, 0.742886, 0.6]})
+    pd.testing.assert_frame_equal(X_t, X_expected)
+
+    encoder = TargetEncoder(smoothing=10)
+    encoder.fit(X, y)
+    X_t = encoder.transform(X)
+    X_expected = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                               'col_2': [2, 1, 1, 1, 1],
+                               'col_3': [0.686166, 0.686166, 0.686166, 0.686166, 0.6]})
+    pd.testing.assert_frame_equal(X_t, X_expected)
+
+    encoder = TargetEncoder(smoothing=100)
+    encoder.fit(X, y)
+    X_t = encoder.transform(X)
+    X_expected = pd.DataFrame({'col_1': [1, 2, 1, 1, 2],
+                               'col_2': [2, 1, 1, 1, 1],
+                               'col_3': [0.676125, 0.676125, 0.676125, 0.676125, 0.6]})
```

Comment:
Interesting! Increasing the smoothing by a factor of 10 barely changed the estimate compared to smoothing=10.